runtime/lua/vim/diagnostic.lua

```diff
@@ -59,8 +59,22 @@ local bufnr_and_namespace_cacher_mt = {
   end,
 }
 
-local diagnostic_cleanup = setmetatable({}, bufnr_and_namespace_cacher_mt)
-local diagnostic_cache = setmetatable({}, bufnr_and_namespace_cacher_mt)
+local diagnostic_cache = setmetatable({}, {
+  __index = function(t, bufnr)
+    if not bufnr or bufnr == 0 then
```

Comment:
Is it even possible for `bufnr` to be nil here? If not, let’s remove the useless `not bufnr` check

Reply:
Tbh I don't actually know?